networkdb/networkdb.go

```diff
@@ -436,8 +436,7 @@ func (nDB *NetworkDB) deleteNetworkEntriesForNode(deletedNode string) {
 	nDB.Unlock()
 }
 
-func (nDB *NetworkDB) deleteNodeNetworkEntries(nid, node string) {
-	nDB.Lock()
+func (nDB *NetworkDB) deleteNodeNetworkEntries(nid, node string, flush bool) {
```

Comment:
Also do we need the `flush` flag ?
If it is used to identify self, cant we simply check for `node == nDB.config.NodeName` ?